### PR TITLE
[SPARK-35120][INFRA] Guide users to sync branch and enable GitHub Actions in their forked repository

### PR DESCRIPTION
**Pull request by HyukjinKwon**
### What changes were proposed in this pull request?

This PR proposes to add messages when the workflow fails to find the workflow run in a forked repository, for example as below:

**Before**

Screen Shot 2021-04-19 at 9 41 52 PM

Screen Shot 2021-04-19 at 9 42 00 PM

**After**

Screen Shot 2021-04-19 at 9 25 32 PM

Screen Shot 2021-04-19 at 9 23 13 PM
(typo `foce` in the image was fixed)

See this example: https://github.com/HyukjinKwon/spark/runs/2380644793

### Why are the changes needed?

To guide users to enable Github Actions in their forked repositories (and sync their branch to the latest `master` in Apache Spark).

### Does this PR introduce _any_ user-facing change?

No, dev-only.

### How was this patch tested?

Manually tested in:
- https://github.com/HyukjinKwon/spark/pull/47
- https://github.com/HyukjinKwon/spark/pull/46

### CODE REVIEW

File: .github/workflows/update_build_status.yml [hunk 1]
```diff
@@ -74,7 +74,8 @@ jobs:
                           check_run_id: cr.id,
                           output: cr.output,
                           status: run.data.status,
-                          conclusion: run.data.conclusion
+                          conclusion: run.data.conclusion,
+                          details_url: run.data.details_url
```
Comment (HyukjinKwon) [inline .github/workflows/update_build_status.yml:78]: It links "View more details on GitHub Actions" to the workflow run in forked repository:
Screen Shot 2021-04-19 at 9 16 26 PM

This is unrelated minor change.

File: .github/workflows/notify_test_workflow.yml [hunk 2]
```diff
@@ -57,34 +57,63 @@ jobs:
...
-              + context.payload.pull_request.head.repo.full_name
-              + '/actions/runs/'
-              + runID
 
             const name = 'Build and test'
             const head_sha = context.payload.pull_request.head.sha
-            const status = 'queued'
+            let status = 'queued'
+
+            if (runs.data.workflow_runs.length === 0) {
```
Comment (HyukjinKwon) [inline .github/workflows/notify_test_workflow.yml:65]: Screen Shot 2021-04-19 at 9 25 32 PM
Screen Shot 2021-04-19 at 9 23 13 PM

**Comment by HyukjinKwon** [comment]
cc @dongjoon-hyun, @gengliangwang and @potiuk FYI

**Review by potiuk — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Thanks @potiuk!

**Review by dongjoon-hyun — APPROVED** [review]
Thank you for adding this guideline.

**Review by gengliangwang — APPROVED** [review]

**Comment by dongjoon-hyun** [comment]
Merged to master.
